src/evolve_momentum.cxx

```diff
@@ -211,8 +224,13 @@ void EvolveMomentum::finally(const Options &state) {
   //  - Density floor should be consistent with calculation of V
   //    otherwise energy conservation is affected
   //  - using the same operator as in density and pressure equations doesn't work
-  ddt(NV) -= AA * FV::Div_par_fvv<hermes::Limiter>(Nlim, V, fastest_wave, fix_momentum_boundary_flux);
-  
+
+  if (isMMS) {
```

Comment:
This feels a bit wrong, to use a different method in MMS testing than in production. The `Div_par_fvv` operator converges for non-FCI, so the issue must be in the FCI branch of `Div_par_fvv`.